oidc.go

```diff
@@ -230,6 +231,22 @@ func (p *Provider) UserInfo(ctx context.Context, tokenSource oauth2.TokenSource)
 		return nil, fmt.Errorf("%s: %s", resp.Status, body)
 	}
 
+	if strings.EqualFold(resp.Header.Get("Content-Type"), "application/jwt") {
+		rawUserInfo := string(body)
+		payload, err := parseJWT(rawUserInfo)
```

Comment:
parseJWT is used as part of ID token validation so we can verify the claims before having to verify the signature (which is potentially more expansive). Since you're not doing that here, this if block can just be

```
if strings.EqualFold(resp.Header.Get("Content-Type"), "application/jwt") {
		payload, err := p.remoteKeySet.VerifySignature(ctx, string(body))
		if err != nil {
			return nil, fmt.Errorf("oidc: invalid userinfo jwt signature %v", err)
		}
		body = payload
}
```

Reply:
yep, correct.

I've played with this for a bit and assumed that this might be useful for scenarios with invalid JWTs - to fail right away without doing the verification.

But I guess this is too much optimization.
I'll remove it.